# STAT 207 Group Lab Assignment 7 - [10 total points]

## Confidence Intervals

<hr>

## <u>Lab Grading</u>:

Should we grade your submission?  If not, write the netID of the submission to be graded.  (Note: We will only grade one assignment per group, and we'll pick the first one that says we should grade that submission.  We will assign the same grade to all team members.)

*For example*, you might respond: **grade this submission** or **my submission is under netID jdeeke**

mdm16

If you said **my submission is under netID** above, we will not read any more of your lab submission.

If you said **grade this submission** above, who worked with you on this submission?  Write both their **names** and **netIDs**.  Also, discuss and record the proportion of your group that has seen a meteor shower or a "shooting star" before.

<hr>

## <u>Purpose</u>:
You should work in groups of 2-3 on this report (not working in groups without permission will result in a point deduction). The purpose of this group lab assignment is to generate two confidence intervals.
<hr>

## <u>Assignment Instructions</u>:

### Group Roles

Suggested and specified roles are provided below: 

#### Groups of 2

* **Driver**: This student will type the report.  While typing the report, you may be the one who is selecting the functions to apply to the data.
* **Navigator**: This student will guide the process of answering the question.  Specific ways to help may include: outlining the general steps needed to solve a question (providing the overview), locating examples within the course notes, and reviewing each line of code as it is typed.

#### Groups of 3

* **Driver**: This student will type the report.  They may also be the one to select the functions to apply to the data.
* **Navigator**: This student will guide the process of answering the question.  They may select the general approach to answering the question and/or a few steps to be completed along the way. 
* **Communicator**: This student will review the report (as it is typed) to ensure that it is clear and concise.  This student may also locate relevant examples within the course notes that may help complete the assignment.

<hr>

### Imports

In [1]:
#Run this
import pandas as pd                    # imports pandas and calls the imported version 'pd'
import matplotlib.pyplot as plt        # imports the package and calls it 'plt'
import seaborn as sns                  # imports the seaborn package with the imported name 'sns'
sns.set()  

## Case Study: Near Earth Objects

We'd like to learn more about **all** objects that pass by the earth closely, in the past or in the future (this is our population of interest).  NASA has a database on near-Earth objects (NEOs) available from their website: https://cneos.jpl.nasa.gov/ca/.  We have prepared and provided for you this dataset for objects within the past year with a passing distance of less than 0.1 au.  The data was downloaded on October 5, 2025.

Variables in the data include:

- **CA DistanceNominal (au)**: the most likely distance between the object and Earth (measured between the center of each object, measured in astronomical units)
- **CA DistanceMinimum (au)**: the minimum possible distance between the object and Earth (measured between the center of the object, measured in astronomical units)
- **V relative(km/s)**: object velocity relative to Earth during its close-approach
- **V infinity(km/s)**: object velocity relative to a massless Earth during its close-approach
- **H(mag)**: asteroid absolute magnitude (smaller H implies larger asteroid diameter)
- **Rarity**: how infrequent a close approach is for asteroids of the same size or larger (larger values are less frequent)

Based on the data, you will pick an appropriate variable and question of interest that you will answer using a confidence interval.

### 1. [1 point] Prepare the Data

For this question, you will pick one of the first 5 variables, select a parameter to learn about, and then clean your data.

**a)** Which of the first 5 variables listed above are you interested in learning about?

CA Distance Nominal

**b)** What parameter are you interested in estimating for the variable selected in part **a**?  You will work with this parameter in Question 2.

Define your parameter of interest for a single population (both with a symbol if possible and in writing).

$\mu =$ True mean CA Distance Nominal for all objects that pass by earth closely in the past in future. 

**c)** Read in your data.  There should not be any missing values present in your data.

In [3]:
df = pd.read_csv("Neo Earth Close Approaches.csv")
df.head()


,Object,Close-Approach (CA) Date,CA DistanceNominal (au),CA DistanceMinimum (au),V relative(km/s),V infinity(km/s),H(mag),Diameter,Rarity
0,(2024 SU5),2024-Oct-06 03:33 ± < 00:01,0.07427,0.07374,23.01,23.01,23.6,51 m - 110 m,0
1,(2024 TZ3),2024-Oct-06 04:45 ± < 00:01,0.05751,0.05720,9.83,9.83,25.2,24 m - 53 m,0
2,(2022 SU21),2024-Oct-06 05:04 ± 2_16:19,0.04488,0.00682,21.10,21.10,24.5,33 m - 74 m,0
3,(2024 TD),2024-Oct-06 05:31 ± < 00:01,0.02480,0.02471,10.92,10.91,26.9,11 m - 25 m,0
4,(2024 RO43),2024-Oct-06 06:20 ± < 00:01,0.06554,0.06542,7.72,7.72,24.8,30 m - 66 m,0


**d)** Create a `Rare` variable that records a value of 1 or True if the object is rare (has a `Rarity` value of 1 or more) and a value of 0 or False if the object is not particularly rare (has a `Rarity` value of 0).

In [ ]:
df['Rare'] = (df['Rarity'] >= 1).astype(int)

Rare
0    2430
1     250
Name: count, dtype: int64

### 2. [5 points] Single Population Confidence Interval

**a)** Provide your best estimate for the population parameter based on our available data.

In [8]:
df['CA DistanceNominal (au)'].agg(['mean'])

mean    0.037776
Name: CA DistanceNominal (au), dtype: float64

**b)** Generate a sampling distribution using your available data as a stand-in for the population.

In [46]:
samp = []
for i in range(5000):
    df_samp = df['CA DistanceNominal (au)'].sample(len(df),replace=True)
    samp.append(df_samp.mean())
samp = pd.Series(samp)



**c)** Calculate a 90% confidence interval from your sampling distribution above.

In [47]:
x = samp.quantile(.05)
y = samp.quantile(.95)
print(f"({x},{y})")

(0.03686490690298507,0.038710045708955226)


**d)** Provide an interpretation of your final **confidence interval**.

We are 90% confident that our true population mean of CA Distance Nominal lies between .0368649 au and .03871004 au.

**e)** Provide an interpretation of the **confidence level**.

a 90% confidence level means that if we repeatedly took samples and created confidence intervals about 90% of those intervals would contain the true population mean of CA Distance Nominal

### 3. [4 points] Two Populations Confidence Interval

We'll now turn to a different parameter of interest that allows us to examine the difference between rare and regular NEOs.  For this parameter of interest, we'll use the difference between the parameter defined in **1b** between those objects that are rare compared to those that are more common.  This will allow us to understand how much larger the value is for rare objects than common ones.

*Note:* it doesn't matter whether you subtract the rare value from the common value or the common value from the rare value, as long as you always subtract the values in the same order throughout this question.

**a)** First, provide your best estimate for the parameter(s) of interest using the available data.

In [26]:
df_common = df[df['Rare'] == 0]
df_rare = df[df['Rare'] >= 1]
w = df_common['CA DistanceNominal (au)'].mean()
v = df_rare['CA DistanceNominal (au)'].mean()
print(f"Parameter for Common (Rareity < 1) : {w},\nParameter for Rare (Rarity >= 1) : {v}")

Parameter for Common (Rareity < 1) : 0.039157526748971196,
Parameter for Rare (Rarity >= 1) : 0.024347920000000002


**b)** Generate a sampling distribution for your parameter of interest.  

To do so, you should first separate your data into a Rare data frame and a Common data frame.  For each repetition inside a `for` loop, you'll take a sample from each of your two sets of data.  You will calculate the statistic from each re-sampled set of data.  Then, subtracting the two statistics allows you to capture your overall statistic of interest.  This statistic of interest is what you should save from each of your repetitions.

In [48]:
sample = []
for i in range(5000):
    df_samp_common = df_common['CA DistanceNominal (au)'].sample(len(df_common), replace=True)
    df_samp_rare = df_rare['CA DistanceNominal (au)'].sample(len(df_rare), replace=True)
    sample.append(df_samp_common.mean() - df_samp_rare.mean())
sample = pd.Series(sample)


**c)** Pick a confidence level other than 90%, and report that number.  Then, generate the corresponding confidence interval for our parameter of interest, corresponding to the difference between the objects that are rare compared to those that are more common.

95% confidence level

In [49]:
l = sample.quantile(.025)
u = sample.quantile(.975)
print(f"({l},{u})")

(0.011261139971193417,0.01834060087654319)


**d)** Does it seem plausible that there is no difference between the rare and common NEOs for your parameter of interest (corresponding to a value of 0 for your parameter of interest)?  Briefly explain.

Since 0 is not in the confidence interval it is no plausible that there is no difference between the rare and common NEO for CA Distance Nominal.

### 4. AI Acknowledgement

Our course policy is that you should write all of your own interpretations and other narrative answers (phrases or sentences) yourself without the assistance of AI.  AI should not be used on Questions 1a, 1b, 2d, 2e, and 3d of this assignment.  

You may use AI to help guide your code, although you should write all of your own code yourself (not copy-paste from another source) and you should cite your use of AI.  I would encourage you to try to generate any necessary code yourself first using course resources and using AI as a debugging tool if/when you reach an error that you can't figure out or to help you perform any coding tasks that are more advanced than we've demonstrated during class (intended only for projects).  

Did you use AI on this assignment?  Did you use other resources outside of our course-provided resources on this assignment?

No AI was used

If you used AI or other resources, answer the following questions to cite your usage.

- Which AI and/or resources did you use?
- What prompts did you ask it?
- How did you integrate the responses into your assignment?  Specifically, which questions or parts are associated with this usage?

Note: answering these three questions are enough for our course but may not be enough for a different course or context.

N/A